File: MyNode.java
package Ex2_code;

import Ex2_code.api.GeoLocation;
import Ex2_code.api.NodeData;

public class MyNode implements NodeData {

    int id;
    MyGeoLocation gl;
    double weight;
    String remark;
    int tag;

    public MyNode (int id, MyGeoLocation gl){
        this.id=id;
        this.gl=gl;
        this.weight=0;
        this.remark=null;
        this.tag=0;
    }

    @Override
    public int getKey() {
        return id;
    }

    @Override
    public GeoLocation getLocation() {
        return gl;
    }

    @Override
    public void setLocation(GeoLocation p) {
        gl.x=p.x();
        gl.y=p.y();
        gl.z=p.z();
    }

    @Override
    public double getWeight() {
        return weight;
    }

    @Override
    public void setWeight(double w) {
        weight=w;
    }

    @Override
    public String getInfo() {
        return remark;
    }

    @Override
    public void setInfo(String s) {
        remark=s;
    }

    @Override
    public int getTag() {
        return tag;
    }

    @Override
    public void setTag(int t) {
        tag=t;
    }

    public String toString(){
        return ("[id:"+this.id+", pos:"+this.getLocation().toString()+"]");
    }
}

File: Ex3/my_node.py
import random


class my_node:
    def __init__(self, id: int, x: float, y: float, z: float):
        self.id = id
     #   self.x = x
      #  self.y = y
       # self.z = z
        self.weight = 0
        self.tag = 0
        self.pos = {'x': x, 'y': y, 'z': z}


    def get_id(self) -> int:
        return self.id

    def get_location(self) -> dict:
        return f"{','.join(str(x) for x in self.pos.values())}"
        #return self.pos.get('x'),self.pos.get('y'),self.pos.get('z')

    def set_location(self, x: float, y: float, z: float):
        self.pos.update({'x': x})
        self.pos.update({'y': y})
        self.pos.update({'z': z})

    def get_weight(self) -> float:
        return self.weight

    def set_weight(self, w: float):
        self.weight = w

    def get_tag(self) -> int:
        return self.tag

    def set_tag(self, tag: int):
        self.tag = tag

    def __str__(self):
        return f"[id:{self.id}, pos: {','.join(str(x) for x in self.pos.values())}]"

    def __repr__(self):
        return f"[id:{self.id}, pos: {','.join(str(x) for x in self.pos.values())}]"



